DigitalLearningSolutions.Data.Tests/DataServices/PasswordResetDataServiceTests.cs

```diff
@@ -90,24 +110,145 @@ public async Task Does_Not_Match_Reset_Passwords_If_No_Reset_Password_With_Given
+
+            // When
+            await service.RemoveResetPasswordAsync(resetPasswordId);
+
+            // Then
+            var userAfterRemoval = userDataService.GetDelegateUserById(userRef.Id);
+            userAfterRemoval.Should().NotBeNull();
+            userAfterRemoval?.ResetPasswordId.Should().BeNull();
+        }
+
+        [Test]
+        public async Task Removing_reset_hash_from_admin_user_removes_ResetPassword_entity()
+        {
+            using var transactionScope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+
+            // Given
+            var userId = UserTestHelper.GetDefaultAdminUser().Id;
+            var userRef = new UserReference(userId, UserType.AdminUser);
+            var resetPasswordId = await GivenResetPasswordWithHashExistsForUsersAsync(HashNotYetInDb, new[] { userRef });
+
+            // When
+            await service.RemoveResetPasswordAsync(resetPasswordId);
+
+            // Then
+            var matchingResetPasswords = connection.GetResetPasswordById(resetPasswordId);
+            matchingResetPasswords.Should().BeEmpty();
+        }
+
+        [Test]
+        public async Task Removing_reset_hash_from_delegate_user_removes_ResetPassword_entity()
+        {
+            using var transactionScope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+
+            // Given
+            var userId = UserTestHelper.GetDefaultDelegateUser().Id;
+            var userRef = new UserReference(userId, UserType.DelegateUser);
+            var resetPasswordId = await GivenResetPasswordWithHashExistsForUsersAsync(HashNotYetInDb, new[] { userRef });
+
+            // When
+            await service.RemoveResetPasswordAsync(resetPasswordId);
+
+            // Then
+            var matchingResetPasswords = connection.GetResetPasswordById(resetPasswordId);
+            matchingResetPasswords.Should().BeEmpty();
+        }
+
+        /// <summary>
+        /// Adds reset password entity for a list of users.
+        /// </summary>
+        /// <param name="hash">Reset hash.</param>
+        /// <param name="users">Must contain at least 1 user.</param>
+        /// <returns>The id of the reset password entity.</returns>
+        private async Task<int> GivenResetPasswordWithHashExistsForUsersAsync(string hash, IEnumerable<UserReference> users)
+        {
+            var userList = users.ToList();
+
+            var firstUser = userList.First();
+
+            service.CreatePasswordReset
+                (new ResetPasswordCreateModel(DateTime.UtcNow, hash, firstUser.Id, firstUser.UserType));
+
+            var resetPasswordId = await GetResetPasswordIdByHashAsync(hash);
+
+            foreach (var user in userList.Skip(1))
+            {
+                await GivenUserHasResetPasswordIdAsync(resetPasswordId, user);
+            }
+
+            return resetPasswordId;
+        }
+
+        private async Task<int> GetResetPasswordIdByHashAsync(string hash)

```

Comment:
Given that we have the test helpers now, should we move these queries across too?